onnxruntime/core/providers/partitioning_utils.cc

```diff
@@ -354,6 +346,10 @@ std::unique_ptr<ComputeCapability> MakeComputeCapability(const GraphViewer& grap
   meta_def->status = ONNX_NAMESPACE::EXPERIMENTAL;
 
   for (const auto& input : ordered_subgraph_inputs) {
+    if (drop_constant_initializers && graph_viewer.IsConstantInitializer(input->Name(), true)) {
```

Comment:
is dropping all constant initializers here too broad? e.g., what if there is a constant initializer which is an input that the CoreML EP does not require to be constant, and thus does not call AddInitializerToSkip() for?

Reply:
Is that a valid use case though? If it's constant why would you pass it in vs. giving coreml the opportunity to have it on the required device pre-execution?

Reply:
There are 2 potential future refinements here
* Provide a mechanism to allow the EP to specify the set of initializers to drop if this is too broad
* Provide a mechanism to remove initializers that become unused when an ORT format model is involved so that we can reclaim that memory.

Reply:
Implemented the second as it simplified code for session state finalization and results in the initializers being able to be freed with an ORT format model. 

Can consider something to allow not dropping all constant initializers if we see a use-case where that is required.

Reply:
ah, I forgot that the CoreML EP copies used initializers over to the model when building it. this seems reasonable now.